msal-dotnet-articles/advanced/monitoring.md

```diff
@@ -5,44 +5,151 @@ description: "How to use the metrics provided by MSAL.NET to monitor your applic
+- `RegionDetails` - the details about the region used to make call, such as the region used and any auto-detection error.
 
-`CacheRefreshReason` - specifies the reason for fetching the access token from the identity provider. See <xref:Microsoft.Identity.Client.CacheRefreshReason>. Use in conjunction with `TokenSource`.
+    > [!NOTE]
+    > Regionalization is available only to internal Microsoft applications.
 
-`TokenEndpoint` - the actual token endpoint URI used to fetch the token. Useful to understand how MSAL resolves the tenant in silent calls and the region in regional calls.
+## OpenTelemetry
 
-> [!NOTE]
-> Regionalization is available only to internal Microsoft applications for now.
+Starting with MSAL 4.58.0, the library supports [OpenTelemetry](https://opentelemetry.io/) - a set of APIs that enable instrumentation, generation, and collection of telemetry data in a consistent and standardized manner. To get started, ensure that you;
 
-`RegionDetails` - the details about the region used to make call, such as the region used and any auto-detection error.
+1. Install the latest version of [MSAL.NET](https://www.nuget.org/packages/Microsoft.Identity.Client).
+1. Add the [OpenTelemetry](https://www.nuget.org/packages/OpenTelemetry#readme-body-tab) package dependency to your project.
+1. Add an exporter dependency, that allows you to export logs, for example, the [Console exporter for OpenTelemetry.NET](https://www.nuget.org/packages/OpenTelemetry.Exporter.Console/1.7.0-alpha.1).
 
-> [!NOTE]
-> Regionalization is available only to internal Microsoft applications for now.
+>[!NOTE]
+>While the console exporter is a good start for local debugging and diagnostics, it's not the best choice for production-deployed applications. We recommend checking out the [official exporter documentation](https://opentelemetry.io/docs/instrumentation/net/exporters/) to learn more about available options. If you are hosting applications on Azure, you may consider ingesting OpenTelemetry data in [Azure Data Explorer](/azure/data-explorer/open-telemetry-connector?tabs=command-line) or [Azure Monitor](/azure/azure-monitor/app/opentelemetry-enable?tabs=aspnetcore).
+
+In your application initialization code, prior to bootstrapping the MSAL authentication client (e.g., `PublicClientApplication` or `ConfidentialClientApplication`), declare a new `meterProvider` instance, using the following code.
+
+```csharp
+using var meterProvider = Sdk.CreateMeterProviderBuilder()
+    .AddMeter("MicrosoftIdentityClient_Common_Meter")
+    .AddConsoleExporter()
+    .Build();
+```
+
+This will initialize the meter provider and use the built-in MSAL.NET meter (`MicrosoftIdentityClient_Common_Meter`) that captures a series of counters and histograms. When a console exporter is used, you should see the output being piped directly in the terminal:
+
+![Example of OpenTelemetry outputting metrics to the terminal](../media/msal-net-logging/opentelemetry-logging.gif)
```

Comment:
nit: On smaller screens the text is a bit unreadable, I just see that something's printed to the console. Maybe just add a screenshot of what the printed metrics data looks like?